pyartcd/pyartcd/pipelines/gen_assembly.py

```diff
@@ -0,0 +1,233 @@
+                    if k not in b:
+                        # move new entry to the beginning
+                        c.move_to_end(k, last=False)
+                return c
+            pr = await self._create_or_update_pull_request(lambda releases_yaml: _merge(assembly_definition, releases_yaml))
+
+            # Sends a slack message
+            message = f"Hi @release-artists , please review assembly definition for {self.assembly}: {pr.html_url}"
+            await self._slack_client.say(message, slack_thread)
+        except Exception as err:
+            error_message = f"Error generating assembly definition: {err}\n {traceback.format_exc()}"
+            self._logger.error(error_message)
+            await self._slack_client.say(error_message, slack_thread)
+            raise
+
+    async def _get_nightlies(self):
+        """ Get nightlies from Release Controllers
+        :param release: release field for rebase
+        :return: NVRs
+        """
+        cmd = [
+            "doozer",
+            "--group", self.group,
+            "--assembly", "stream",
+        ]
+        if self.arches:
+            cmd.append("--arches")
+            cmd.append(",".join(self.arches))
+        cmd.append("get-nightlies")
+        if self.allow_pending:
+            cmd.append("--allow-pending")
+        if self.allow_rejected:
+            cmd.append("--allow-rejected")
+        if self.allow_inconsistency:
+            cmd.append("--allow-inconsistency")
+        for nightly in self.nightlies:
+            cmd.append(f"--matching={nightly}")
+
+        _, out, _ = await exectools.cmd_gather_async(cmd, stderr=None, env=self._doozer_env_vars)
+        return out.strip().split()
+
+    async def _gen_assembly_from_releases(self, candidate_nightlies: Iterable[str]) -> OrderedDict:
+        """ Run doozer release:gen-assembly from-releases
+        :return: Assembly definition
+        """
+        cmd = [
+            "doozer",
+            "--group", self.group,
+            "--assembly", "stream",
+        ]
+        if self.arches:
+            cmd.append("--arches")
+            cmd.append(",".join(self.arches))
+        cmd.append("release:gen-assembly")
+        cmd.append(f"--name={self.assembly}")
+        cmd.append("from-releases")
+        for nightly in candidate_nightlies:
+            cmd.append(f"--nightly={nightly}")
+        if self.custom:
+            cmd.append("--custom")
+        else:
+            if self.in_flight:
+                cmd.append(f"--in-flight={self.in_flight}")
+            for previous in self.previous_list:
+                cmd.append(f"--previous={previous}")
+            if self.auto_previous:
+                cmd.append("--auto-previous")
+        _, out, _ = await exectools.cmd_gather_async(cmd, stderr=None, env=self._doozer_env_vars)
+        return yaml.load(out)
+
+    async def _create_or_update_pull_request(self, fn: Callable[[OrderedDict], OrderedDict]):
```

Comment:
Maybe `fn` is not the best parameter name, how about `merge_function` or the like?

Reply:
I removed the callback parameter.